### ISSUE TITLE
[FEATURE] Display choice one way and pass as variable in another way

### ISSUE THREAD

**Issue by rolaca11** [jenkinsci/rest-list-parameter-plugin#81]
**Is your feature request related to a problem? Please describe.**
In my REST API I receive a list of JSON objects, of which I need multiple properties. Currently, if I want this option, I have to have the option appear as a JSON string.

**Describe the solution you'd like**
 I'd like the option for this JSON object to appear not as a JSON string, but for example the name property of that object. So if the response from my API is: 
```
[
  {
    "name": "My name",
    "age": 10,
    "profession": "dev"
  },
  {
    "name": "Other name",
    "age": 12,
    "profession": "dr"
  },
```

If I select the 1st option, I'd like my parameter value to be 
```
  {
    "name": "My name",
    "age": 10,
    "profession": "dev"
  }
```

I'd also like the options displayed as "My name" and "Other name", or maybe even "My name, 10" and "Other name, 12".

**Describe alternatives you've considered**
Extend the JSONPath expression logic to allow concatenation of multiple properties. This way, I can patch together the data I need in my pipeline, like "name|age|profession", which then I can split in the pipeline.

**Comment by h1dden-da3m0n** [comment]
hmmm, very interesting use-case.
Thank you for this feature request/idea! I am a bit occupied with work these days, but I sure find your request tempting to experiment with.

The easiest initial impl that comes to my mind is 'flatting' the json object and presenting that. (that would also be how the pipeline would receive the json object) 
form your example `{ "name": "...", "age": ..., "profession": "..." }` (however that is likely anyway how the json gets recived in the first place, cause who in their right mind sends pretty printed json)
But since that indeed would be a mess to select I also like the idea of a 'second' JSONPath for the select value, however how to realise that in a clean way with a JSONPath, XPath and 'simple' single value select option might be a bit more tricky.

Regardless, very neat proposal and I will see what I can realise in due time.

Disclaimer: If you or someone else wants to offer a PR before I have time to work on this, please feel more than welcome to.

**Comment by rolaca11** [comment]
I just forked the repository and am testing out a couple of ideas about this feature, I'll report back if I have any results

**Comment by h1dden-da3m0n** [comment]
Awesome!
Thank you

**Comment by h1dden-da3m0n** [comment]
released as part of v1.5.0
